server/public/js/chart.js

```diff
@@ -1,33 +1,201 @@
-var core1Stats = ['core1'];
-var chart = c3.generate({
-  bindto: '#chart',
-  data: {
-    columns: [
-      core1Stats
-    ],
-    types: {
-      core1: 'area'
-    }
-  }
+'use strict';
+
+class CpuStatsModel {
+    constructor() {
+        this.data = [];
+        this.keys = ['date'];
+        this.maxAge = -1;
+    }
+
+    initKeys(callback) {
+        $.getJSON('cpu-stats/latest',
+            data => {
+                const processorUsage = data.response[0].perProcessorUsage;
+                for (var i = 0; i < processorUsage.length; i++) {
+                    this.keys.push('core' + i);
+                }
+                callback(this.keys);
+            }
+        );
+    }
+
+    update(callback) {
+        $.getJSON('cpu-stats/latest', data => {
+            const resp = data.response[0];
+            const date = new Date(parseInt(resp.timeStamp.$numberLong));
+            this.data.push([date].concat(resp.perProcessorUsage));
```

Comment:
The previous code had an issue where the data array 'core1Stats' is unbounded in size. Does this code also have the issue?

The fix would be to pop elements from the array when they're not needed.